### PR TITLE
Port random affine, rotate, perspective and to_grayscale to pytest

### PR DESCRIPTION
**Pull request by vivekkumar7089**
Refactor Group D as mentioned in #3987 
- group D
  - test_random_affine -- split this into different parametrized functions (one for shear, one for scale, etc.)
  - test_random_rotate -- parametrize over all loop variables
  - test_random_perspective -- parametrize over all loop variables
  - test_to_grayscale -- parametrize over meth_kwargs

### CODE REVIEW

**Review by NicolasHug — COMMENTED** [review]
Thanks @vivekkumar7089 ! a few comments but this looks good

File: test/test_transforms_tensor.py [hunk 1]
```diff
@@ -718,5 +626,122 @@ def setUp(self):
...
+    _test_transform_vs_scripted_on_batch(transform, s_transform, batch_tensors)
+
+    with get_tmp_dir() as tmp_dir:
+        s_transform.save(os.path.join(tmp_dir, "t_perspective.pt"))
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('tol', [1.0 + 1e-10])
+@pytest.mark.parametrize('num_output_channels', [0, 1, 3])
+def test_to_grayscale(device, num_output_channels, tol):
```
[the remark this answers is not in the document]
Comment (vivekkumar7089) [reply test/test_transforms_tensor.py:730]: Thanks, @NicolasHug !! I will take care of all comments which you have made.

File: test/test_transforms_tensor.py [hunk 2]
```diff
@@ -718,5 +626,122 @@ def setUp(self):
...
+    transform = T.RandomRotation(
+        degrees=degrees, interpolation=interpolation, expand=expand, center=center, fill=fill
+    )
+    s_transform = torch.jit.script(transform)
+
+    _test_transform_vs_scripted(transform, s_transform, tensor)
+    _test_transform_vs_scripted_on_batch(transform, s_transform, batch_tensors)
+
+    with get_tmp_dir() as tmp_dir:
+        s_transform.save(os.path.join(tmp_dir, "t_random_rotate.pt"))
```
[the remark this answers is not in the document]
Comment (vivekkumar7089) [reply test/test_transforms_tensor.py:702]: Hi @NicolasHug !! I have changed everything, the rest of this. Actually, I couldn't find a way to reuse `s_transform` outside of its parent function. Could you please help me to solve this problem?

File: test/test_transforms_tensor.py [hunk 3]
```diff
@@ -718,5 +626,122 @@ def setUp(self):
...
+
+    _test_transform_vs_scripted(transform, s_transform, tensor)
+    _test_transform_vs_scripted_on_batch(transform, s_transform, batch_tensors)
+
+    return s_transform
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+def test_random_affine(device):
+    s_transform = _test(device, degrees=0.0)
```
Comment (NicolasHug) [inline test/test_transforms_tensor.py:643]: I realize this was already present, but we don't need to get `s_transform` from `_test()`. `_test()` doesn't need to return anything and we can just declare `s_transform` here as `s_transform = torch.jit.scriptT.RandomAffine(...)`

(where `...` may actually be empty)

File: test/test_transforms_tensor.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (NicolasHug) [reply test/test_transforms_tensor.py:702]: I made a comment above, hopefully this helps? LMK!

[the remark this answers is not in the document]
Comment (vivekkumar7089) [reply test/test_transforms_tensor.py:702]: It means here I can define a function.
```
def test_random_rotate_save():
    s_transform = torch.jit.scriptT.RandomAffine(...)
    with get_tmp_dir() as tmp_dir:     
         s_transform.save(os.path.join(tmp_dir, "t_random_rotate.pt")
```
And also in a similar manner in the function `test_random_perspective`.
@NicolasHug

[the remark this answers is not in the document]
Comment (NicolasHug) [reply test/test_transforms_tensor.py:702]: Yes this sounds like the right way here!

[the remark this answers is not in the document]
Comment (vivekkumar7089) [reply test/test_transforms_tensor.py:702]: Hello @NicolasHug !!
`s_transform = torch.jit.script(T.RandomAffine())`
and
`s_transform = torch.jit.script(T.RandomRotation())`
both are giving same error -: TypeError: __init__() missing 1 required positional argument: 'degrees'
while 
`s_transform = torch.jit.script(T.RandomPerspective())` is working fine.
Can we fix some degree here? (in above cases)

[the remark this answers is not in the document]
Comment (NicolasHug) [reply test/test_transforms_tensor.py:705]: yes we can just pass degree=0

**Comment by NicolasHug** [comment]
I took care of a minor pep8 issue, thanks a lot @vivekkumar7089 !

**Pull request merged**